README.md

```diff
@@ -92,3 +92,32 @@ The path of the project is always relative to a given `tsconfig.json`.
 If no `tsconfig.json` is given `ts-arch` tries to find one in a parent
 folder, e.g. if your `tsconfig.json` is in the same folder as your `src` folder, then all the paths 
 begin with `src/...`
+
+## Dependency checks on nx monorepositories
+
+`ts-arch` supports dependency checks on nx monorepositories. It reads the project graph
+and makes it accessible for the slices api.
+
+The following example illustrates this:
+
+```typescript
+import "tsarch/dist/jest"
+import {slicesOfProject} from "tsarch" 
+import * as path from "path"
+
+describe("architecture", ()=> {
+    jest.setTimeout(60000);
+
+    it('the architecture adheres to the diagram', async () => {
+    	const diagramLocation = path.resolve('docs', 'components.puml');
+    
+    	const rule = await slicesOfNxProject()
+            .asTheyAre(true)
+    		.should()
```

Comment:
This API is not especially clear, particularly given that `asTheyAre(true)` and `identity(true)` are not true identity/homomorphic functions, as they perform filtering. I would instead suggest `slicesOfNxProject().withoutExternals().should()` in this API, and `SliceConditionBuilder` could extend `SlicesShouldCondition` so that, when externals are not filtered out, `slicesOfNxProject().should()` can be used directly.

Please then also replace `identity(true)` with `removeExternals`, and `identity(false)` with `identity`.

Reply:
I also think that the identity should be an isomorphism. I am not sure what the value of "if I choose `withoutExternal once I don't want to see it again in the auto completion" is. I thus removed `SlicesShouldCondition` completely. I can restore it if we find this useful.

Reply:
Thanks, I fixed that.